Add several new optional URL tags 2

ffesti opened the pull request:
Reworked #2045

pmatilai commented on docs/manual/tags.md:
```diff
@@ -61,6 +61,7 @@ Cookie             | 1094 | string       | An opaque value for tracking packages
 Distribution       | 1010 | string       | Distribution name.
 Disttag            | 1155 | string       | Distribution acronym.
 Disturl            | 1123 | string       | Distribution specific URL of the package.
+Downloadurl        | 5101 | string       | URL to list of upstream releases in machine readable form if possible
```
I quite dislike both the tag name and the docs. The name because it's so very ambiguous and even misleading, "downloading" is something *users* do, eg "where do I (re)download this very rpm package", when this is about upstream sources and robots. The doc because it's equally ambiguous and I wouldn't have a clue what to put in there: WTF does "list of releases in machine readable form, if possible" really mean? Is a list of releases "1.2", "1.3" or is it supposed to contain the actual source tarball or what? 

If it's about sources then lets say it both in the tag, and it's doc. I don't have good suggestions offhand, but to throw some ideas around: SourceURL would be fine but then it's easily confused with Source: and could be expected to have related semantics, so not good. ReleaseURL may be closer to describing the intent, especially when documented, but dunno.

ffesti replied:
Yeah, the issue is we have used up all words in that area for other things already: Release, Source, ...

What about `UpstreamReleases` "Location/URL to check for newer releases from upstream. If possible one that can be checked automatically to notify the packager." ?

pmatilai replied:
UpstreamReleases is probably the best so far, simply because it's most to the actual point.
That doc gives the impression that *rpm* may notify the packager, which wont be the case. Maybe something more in the lines of "URL to check for newer releases from upstream. The purpose of this tag is to allow external software such as \<wellknownexample\> to check for newer upstream releases and notify the packager."

ffesti replied:
Shortened the description a bit but kept the content more or less the same.

pmatilai replied:
I wonder if that tag should actually have an URL in the end. It's a bit long as it is, but it'd make the point more obvious...

pmatilai commented:
/me also notes these v2 PR's harmful: all the initial discussion context and participants are lost. Not that I know how these should really be handled - a v2 is fine when there's nothing more to discuss, but I don't think that's the case here :thinking:

ffesti commented:
> /me also notes these v2 PR's harmful: all the initial discussion context and participants are lost. Not that I know how these should really be handled - a v2 is fine when there's nothing more to discuss, but I don't think that's the case here thinking

I agree this is less than ideal. Otoh I don't really see another option if the branch of the PR is not writable and the changes are not done by the contributor. I added some lines in #2084 to encourage people to keep their branches writable.

pmatilai commented:
Tag names aside, the other issue I also raised in the original PR is that I don't think these belong to the --info output, that thing is long enough as it is. Couple of options I could think of:
- add some --extra-info type switch where to put this type of info
- emit these tags only in verbose mode (`%|VERBOSE?{..}}` can be used for this)

ffesti commented:
OK, removed the changes to --info. Instead I added a new `--Info` or `-I` query alias that add the new tags and VCS (and probably more in the future). Idea being that this is easier any type and remember that some `--extra-info` parameter. But I am open for better suggestions.

pmatilai commented:
Info vs info is only going to be confusing as hell. The important stuff needs to be in --info and for the rest, whether it's fast to type is not that important. --extra-info is a stupid name of course, it'd need to be something more descriptive. But like said, adding more output on -v seems like an entirely reasonable option to me.

pmatilai commented:
FWIW, we could also just add the tags here, a bot that looks for upstream releases is better off doing a --qf than parsing some --info type output. And translation url is not something the average user needs to be able to quickly and easily see either. 

(I'm finding myself thinking of moving the --info output to C code to be able to better handle various things, but that's outside the scope here)

ffesti commented:
Well, when stepping back a bit I wonder whether we can actually remove things from the normal --info if we have a long/extended info else where. Candidates might be:
```
Size
License
Signature
Source RPM
Build Date
Build Host
Packager
Vendor
Bug URL
```
ofc one can argue with any one of that.

pmatilai commented:
And that's why I'd rather just get the tags in, and argue about --info output separately :sweat_smile:

ffesti commented:
Here you go!

stanislav-brabec commented:
Thanks for the improvement of my suggestion. I am happy with the new patch.

There is just a single thing open. At least the TranslationURL (but also UpstreamReleases) may need more that one URL. The original discussion says that the technically correct tag array is not optimal for use, and use of a plain string and a delimiter is a better way to do it.

So the last question is: Should the comment propose a delimiter? Which character? I guess it could be say ";". It is rarely used in project URLs, and if it is, we can use %3B.
